front/app/containers/EventsShowPage/EventTopBarMobile.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useCallback } from 'react';
+
+// hooks
+import useProjectById from 'api/projects/useProjectById';
+
+// i18n
+import useLocalize from 'hooks/useLocalize';
+
+// components
+import GoBackButtonSolid from 'components/UI/GoBackButton/GoBackButtonSolid';
+import { useBreakpoint, Box } from '@citizenlab/cl2-component-library';
+
+// routing
+import clHistory from 'utils/cl-router/history';
+
+// styling
+import styled from 'styled-components';
+import { media, colors } from 'utils/styleUtils';
+
+const Container = styled.div`
+  flex: 0 0 ${(props) => props.theme.mobileTopBarHeight}px;
+  height: ${(props) => props.theme.mobileTopBarHeight}px;
+  background-color: #fff;
+  border-bottom: solid 1px ${colors.coolGrey300};
+  position: sticky;
+  top: 0;
+  z-index: 2000;
+`;
+
+const TopBarInner = styled.div`
+  height: 100%;
+  padding-left: 15px;
+  padding-right: 15px;
+  position: relative;
+  display: flex;
+  align-items: center;
+  justify-content: space-between;
+
+  ${media.desktop`
+    padding-left: 30px;
+    padding-right: 30px;
+  `}
+`;
+
+interface Props {
+  projectId?: string;
+}
+
+const EventTopBarMobile = ({ projectId }: Props) => {
+  const { data: project } = useProjectById(projectId);
+  const isSmallerThanTablet = useBreakpoint('tablet');
+
+  const localize = useLocalize();
+
+  const handleGoBack = useCallback(() => {
```

Comment:
Have you considered using `clHistory.back()` instead? Benefits:

1. You always go back to the page from where you clicked on the event. In your code it depends on whether the event has a project relationship. So if you would click on the event from e.g. the events overview page, but the event is associated with a project, you would maybe prefer to go back to the events overview page instead of to the project (the current behavior)
2. You return to the scroll position of the previous page

Reply:
Hmm yeah I thought about that, but on the idea page today it doesn't work like that :thinking: If you go to an idea from the all inputs page, the grey back button with the project title will always take you to the project.

If we removed the project title there and had the go back arrow return to the previous page, then we'd just need to add a "View project" button on the event page somewhere else I think so users can still find it. I'll have a chat with Rob about it :) Thanks for raising this!

![image](https://github.com/CitizenLabDotCo/citizenlab/assets/33987955/5fffaff3-50c2-45c0-9d1a-e066cd21575c)